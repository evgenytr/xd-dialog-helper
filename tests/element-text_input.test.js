const textInput = require('../dialog-helper').types.TEXT_INPUT;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

const rendered = textInput.render('dialog', {
    id: 'hr',
    label: 'Hello World',
    value: 'abc'
}, actions);

describe('Text Input Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return the value', () => {
            expect(textInput.value(rendered)).toBe('abc');
        });
    });
    describe('validate()', () => {
        it('should always return true if there\'s no required prop', () => {
            expect(textInput.valid(rendered)).toBeTruthy();
        });

        it('should return false for empty fields if required == true', () => {
            let emptyField = textInput.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: '',
                required: true
            }, actions);
            expect(textInput.valid(emptyField)).toBeFalsy();
        });

        it('should return true for non-empty fields if required == true', () => {
            let emptyField = textInput.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: 'a',
                required: true
            }, actions);
            expect(textInput.valid(emptyField)).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(textInput.type).toMatchSnapshot();
    });
});
