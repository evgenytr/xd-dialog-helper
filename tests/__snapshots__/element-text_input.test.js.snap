// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Input"`;

exports[`Text Input Element render() should render correctly 1`] = `
Object {
  "input": <input
    id="dialog-hr"
    placeholder="Hello World"
  />,
  "props": Object {
    "id": "hr",
    "label": "Hello World",
    "value": "abc",
  },
  "type": Object {
    "render": [Function],
    "type": "Input",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-hr-wrapper"
  >
    <span
      id="dialog-hr-label"
    >
      Hello World
      <br />
    </span>
    <input
      id="dialog-hr"
      placeholder="Hello World"
    />
  </label>,
}
`;
